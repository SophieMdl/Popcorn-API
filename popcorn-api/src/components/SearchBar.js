import React, { Component } from 'react'
import actions from '../actions/file'

class SearchBar extends Component {
    state = {
      searchText: '',
      placeHolder: 'Search'
    }
    handleChange = event => {
      this.setState({ searchText: event.target.value })
    }
    handleOnClick = event => {
      actions.searchMovie(this.state.searchText)
    }
    render () {
      return (
        <div className="row input-field">
          <input className="validate col s8" placeholder={this.state.placeHolder} onChange={this.handleChange} />
          <a className="waves-effect waves-light btn-small" onClick={this.handleOnClick}><i className="material-icons left">search</i></a>
        </div>
      )
    }
}

export default SearchBar
