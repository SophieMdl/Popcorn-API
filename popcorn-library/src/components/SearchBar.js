import React from 'react'

const SearchBar = ({ moviesTitles, searchText, changeTextSearch, searchMovie }) => {
  return (
    <div className="row input-field">
      <form onSubmit={(e) => { e.preventDefault(); searchMovie(searchText) }}>
        <input list="title-movie"
          placeholder='search'
          value={searchText || ''}
          className='col s8'
          onChange={(e) => changeTextSearch(e.target.value, e)}
        />
        <datalist id="title-movie">
          {moviesTitles.map((title, index) =>
            <option key={index} value={title} />
          )}
        </datalist>
        <button className="waves-effect waves-light btn-small">
          <i className="material-icons left">search</i>
        </button>
      </form>
    </div>
  )
}

export default SearchBar
