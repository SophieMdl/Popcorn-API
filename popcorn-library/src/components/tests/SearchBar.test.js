import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import SearchBar from '../SearchBar'

const moviesTitles = [
  'test', 'title2', 'title3'
]

it('renders without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(<SearchBar moviesTitles={moviesTitles} />, div)
  ReactDOM.unmountComponentAtNode(div)
})

it('renders SearchBar html', () => {
  const tree = renderer
    .create(<SearchBar moviesTitles={moviesTitles} searchText='title2' />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
