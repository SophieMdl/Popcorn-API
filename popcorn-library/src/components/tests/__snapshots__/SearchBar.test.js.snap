// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders SearchBar html 1`] = `
<div
  className="row input-field"
>
  <form
    onSubmit={[Function]}
  >
    <input
      className="col s8"
      list="title-movie"
      onChange={[Function]}
      placeholder="search"
      value="title2"
    />
    <datalist
      id="title-movie"
    >
      <option
        value="test"
      />
      <option
        value="title2"
      />
      <option
        value="title3"
      />
    </datalist>
    <button
      className="waves-effect waves-light btn-small"
    >
      <i
        className="material-icons left"
      >
        search
      </i>
    </button>
  </form>
</div>
`;
